src/test/ui/suggestions/as-ref.stderr

```diff
@@ -2,9 +2,10 @@ error[E0308]: mismatched types
   --> $DIR/as-ref.rs:6:27
    |
 LL |   opt.map(|arg| takes_ref(arg));
-   |       ---                 ^^^ expected &Foo, found struct `Foo`
-   |       |
-   |       help: consider using `as_ref` instead: `as_ref().map`
```

Comment:
This is a regression that should be addressed before the PR can be merged. Any clue what lead to it?

Reply:
I am investigating it

Reply:
Perhaps some parent-child relations that the suggestion relies on were broken by the new nodes in HIR?